tools/alara_output_processing/alara_output_processing.py

```diff
@@ -130,57 +168,118 @@ def extract_tables(self):
                 current_table_lines.append(line)
                 if self._is_end_of_table(line):
                     if current_parameter and current_block:
-                        self._parse_table_data(
+                        self.rows.extend(self._parse_table_data(
                             current_table_lines,
                             current_parameter,
                             current_block
-                        )
+                        ))
                     inside_table = False
                     current_table_lines = []
                 continue
+        
+        self.results = ALARADFrame(self.rows)
 
-        if not self.results:
+        if self.results.empty:
             warn(f'Unable to read tables from {self.filepath}')
 
         return self.results
 
-    # ---------- Output ----------
-    def write_csv_files(self):
-        '''
-        Write out all ALARADFrames extracted from parsed ALARA output tables
-            to their own CSV files.
-            
-        Arguments:
-            self (alara_output_processing.FileParser): FileParser object.
-        
-        Returns:
-            None
-        '''
-
-        for key, df in self.results.items():
-            filename = self._sanitize_filename(key) + '.csv'
-            df.to_csv(filename, index=False)
 
 class ALARADFrame(pd.DataFrame):
     '''
     A subclass of pandas.DataFrame specialized for ALARA output.
     '''
+    _metadata = ['is_pivot', 'is_single_var', 'is_single_run']
+    VARIABLES = ['Number Density', 'Specific Activity', 'Total Decay Heat', 'Contact Dose']
+    BLOCKS = ['Interval', 'Zone', 'Material']
+    VARAIBLE_ENUM = {name: i for i, name in enumerate(VARIABLES)}
+    BLOCK_ENUM = {name: i for i, name in enumerate(BLOCKS)}
 
     @property
     def _constructor(self):
         return ALARADFrame
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_pivot = False
+        self.is_single_var = False
+        self.is_single_run = False
+
+    def select_single_var(self, variable):
```

Comment:
Could we combine these methods into a generic filtering method that takes a dictionary of different filters, e.g:
```
def filter_rows(self, filter):
    # take the intersection of the filter keys with the allowable columns
    # perform a filtering of the dataframe with all the provided/allowable columns
    # return that dataframe without pivoting yet
```